node_modules/npm-lifecycle/index.js

```diff
@@ -262,7 +262,7 @@ function runCmd_ (cmd, pkg, env, wd, opts, stage, unsafe, uid, gid, cb_) {
 
   if (customShell) {
     sh = customShell
-  } else if (process.platform === 'win32') {
```

Comment:
This file is in a dependency. To patch this you need to PR this change to [npm-lifecycle](https://github.com/npm/npm-lifecycle).

I think this is the right approach though.  When you PR this to `npm-lifecycle` make sure you include a default value for the `isWindowsShell` option that's compatible.

Reply:
done.